File: Golang/RemoveElement/RemoveElement_test.go
package RemoveElement

import "testing"

func TestRemoveElement(t *testing.T) {

	//Expected:
	//nums: [2, 2, 3, 3] length = 2
	if removeElement([]int{3, 2, 2, 3}, 3) == 2 {
		t.Log("pass")
	} else {
		t.Error("failed")
	}

	//Expected:
	//nums: [3, 3] length = 0
	if removeElement([]int{3, 3}, 3) == 0 {
		t.Log("pass")
	} else {
		t.Error("failed")
	}

	//Expeted:
	//nums: [] length = 0
	if removeElement([]int{}, 0) == 0 {
		t.Log("pass")
	} else {
		t.Error("failed")
	}

	//Expected:
	//nums: [3, 4, 5, 888, 2, 2] length = 4
	if removeElement([]int{2, 3, 4, 5, 888, 2}, 2) == 4 {
		t.Log("pass")
	} else {
		t.Error("failed")
	}

	//Expected:
	//nums: [3, 3] length = 2
	if removeElement([]int{3, 3}, 0) == 2 {
		t.Log("pass")
	} else {
		t.Error("failed")
	}

}

File: Python/RemoveElementTest.py
from unittest import TestCase

from RemoveElement import RemoveElement


class TestRemoveElement(TestCase):
    def test_removeElement(self):
        re = RemoveElement()

        # nums: [2, 2, 3, 3] length = 2
        self.assertTrue(re.removeElement([3, 2, 2, 3], 3) == 2)

        # nums: [3, 3] length = 0
        self.assertTrue(re.removeElement([3, 3], 3) == 0)
